plasmapy/formulary/collisions.py

```diff
@@ -565,21 +570,21 @@ def _replace_nan_velocity_with_thermal_velocity(V, T, m):
         raise utils.PhysicsError("Collisions are not possible with a zero velocity.")
 
     if V is None:
-        return thermal_speed(T, "e-", mass=m)
+        return thermal_speed(T, species, mass=m)
 
     if not np.any(np.isnan(V)):
         return V
 
     if not (np.isscalar(V.value) or np.isscalar(T.value)):
```

Comment:
I think the better way to handle the scalar/vector issue is to just force everything to be a vector at the beginning. E.g. 

```python
if np.isscalar(V.value):
     V = np.array([V.value, ]) * V.unit
```

Then you can treat it like a vector from then on

Reply:
I didn't write this function but I agree there are some modifications that would make it a little cleaner

Reply:
Ah ok, you can also just leave it